snapcraft/tests/test_plugin_maven.py

```diff
@@ -0,0 +1,131 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2016 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import os
+
+from unittest import mock
+
+from snapcraft import tests
+from snapcraft.plugins import maven
+
+
+class MavenPluginTestCase(tests.TestCase):
+
+    def setUp(self):
+        super().setUp()
+
+        class Options:
+            maven_options = []
+
+        self.options = Options()
+
+        patcher = mock.patch('snapcraft.repo.Ubuntu')
+        self.ubuntu_mock = patcher.start()
+        self.addCleanup(patcher.stop)
+
+    def test_schema(self):
+        schema = maven.MavenPlugin.schema()
+
+        properties = schema['properties']
+        self.assertTrue('maven-options' in properties,
+                        'Expected "maven-options" to be included in '
+                        'properties')
+
+        maven_options = properties['maven-options']
+
+        self.assertTrue(
+            'type' in maven_options,
+            'Expected "type" to be included in "maven-options"')
+        self.assertEqual(maven_options['type'], 'array',
+                         'Expected "maven-options" "type" to be "array", but '
+                         'it was "{}"'.format(maven_options['type']))
+
+        self.assertTrue(
+            'minitems' in maven_options,
+            'Expected "minitems" to be included in "maven-options"')
+        self.assertEqual(maven_options['minitems'], 1,
+                         'Expected "maven-options" "minitems" to be 1, but '
+                         'it was "{}"'.format(maven_options['minitems']))
+
+        self.assertTrue(
+            'uniqueItems' in maven_options,
+            'Expected "uniqueItems" to be included in "maven-options"')
+        self.assertTrue(
+            maven_options['uniqueItems'],
+            'Expected "maven-options" "uniqueItems" to be "True"')
+
+    @mock.patch.object(maven.MavenPlugin, 'run')
+    @mock.patch('glob.glob')
+    def test_build(self, glob_mock, run_mock):
+        plugin = maven.MavenPlugin('test-part', self.options)
+        os.makedirs(plugin.sourcedir)
+        glob_mock.return_value = [
+            os.path.join(plugin.builddir, 'target', 'fake')]
```

Comment:
You are not using this value in the test, it seems it's just needed to be able to test the the call to run. To make that clear I usually call the value 'dummy' instead of 'fake'.